bpf/lib/time.h

```diff
@@ -6,22 +6,46 @@
 #include <bpf/ctx/ctx.h>
 #include <bpf/api.h>
 
+#include "static_data.h"
+
 #define NSEC_PER_SEC	(1000ULL * 1000ULL * 1000UL)
-#define NSEC_PER_MSEC	(1000ULL * 1000ULL)
-#define NSEC_PER_USEC	(1000UL)
+
+DECLARE_CONFIG(bool, enable_jiffies, "Use jiffies (count of timer ticks since boot).");
+
+DECLARE_CONFIG(__u32, kernel_hz, "Number of clock ticks per second.");
+ASSIGN_CONFIG(__u32, kernel_hz, 250);
 
 /* Monotonic clock, scalar format. */
-#define bpf_ktime_get_sec()	\
-	({ __u64 __x = ktime_get_ns() / NSEC_PER_SEC; __x; })
-#define bpf_ktime_get_msec()	\
-	({ __u64 __x = ktime_get_ns() / NSEC_PER_MSEC; __x; })
-#define bpf_ktime_get_usec()	\
-	({ __u64 __x = ktime_get_ns() / NSEC_PER_USEC; __x; })
-#define bpf_ktime_get_nsec()	\
-	({ __u64 __x = ktime_get_ns(); __x; })
+static __always_inline __u64 bpf_ktime_get_sec(void)
+{
+	__u64 s = ktime_get_ns() / NSEC_PER_SEC;
+	return s;
+}
+
+static __always_inline __u64 bpf_ktime_get_nsec(void)
+{
+	return ktime_get_ns();
+}
 
 /* Jiffies */
-#define bpf_jiffies_to_sec(j)	\
-	({ __u64 __x = (j) / KERNEL_HZ; __x; })
-#define bpf_sec_to_jiffies(s)	\
-	({ __u64 __x = (s) * KERNEL_HZ; __x; })
+static __always_inline __u64 bpf_sec_to_jiffies(__u64 s)
+{
+	__u64 j = s * CONFIG(kernel_hz);
+	return j;
+}
+
+#define BPF_MONO_SCALER 8
+
+static __always_inline __u64 bpf_mono_now(void)
+{
+	if (CONFIG(enable_jiffies) && CONFIG(kernel_hz) != 1)
+		return jiffies >> BPF_MONO_SCALER;
```

Comment:
It took me a while to figure out what `jiffies` was defined as. Apparently there's a `#define jiffies	jiffies64()` in `bpf/helpers.h`. My 2c: I'd remove that `#define` and turn this line into

```suggestion
		return jiffies64() >> BPF_MONO_SCALER;
```

This makes it clear that the value originates from a bpf helper call. If tests want to sub the bpf helper, they can `#define jiffies64() ...` in the appropriate place like conntrack_test.c is doing.

Reply:
Removed and pushed in a separate commit.